src/core/libmaven/mzAligner.cpp

```diff
@@ -515,9 +521,11 @@ bool Aligner::alignWithObiWarp(vector<mzSample*> samples,
     float minMzRange = 1e9;
     float maxMzRange = 0;
     for (int j = 0; j < refSample->scans.size(); ++j) {
-        for (int k = 0; k < refSample->scans[j]->mz.size(); ++k) {
-            minMzRange = min(minMzRange, refSample->scans[j]->mz[k]);
-            maxMzRange = max(maxMzRange, refSample->scans[j]->mz[k]);
+        if(refSample->scans[j]->mslevel  == 1 ){
+            for (int k = 0; k < refSample->scans[j]->mz.size(); ++k) {
+                minMzRange = min(minMzRange, refSample->scans[j]->mz[k]);
+                maxMzRange = max(maxMzRange, refSample->scans[j]->mz[k]);
+            }
         }
     }
```

Comment:
Although this won't result in any increased efficiency in this case, the whole for-block might be more readable this way:
```C++
    for (const auto scans : refSample->scans) {
        if(scan->mslevel > 1 )
            continue;
        for (const auto mz : scan->mz) {
            minMzRange = min(minMzRange, mz);
            maxMzRange = max(maxMzRange, mz);
        }
    }
```
It has one less level of indentation and uses range based loops. I understand this is not part of the change you were trying to introduce, so you need not implement it this way. This is just a suggestion.